src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,290 @@
+    # Load grant request for device_c in SAS Test Harness
+    grant_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_c['cbsdId'] = device_c['userId']
+    grant_c['operationParam']['maxEirp'] = 20
+    # Load device_d within the PPA to SAS UUT
+    device_d = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_d.json')))
+    device_d['installationParam']['latitude'] = 38.7790
+    device_d['installationParam']['longitude'] = -97.2263
+    # Load grant request for device_d to SAS UUT
+    grant_d = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_d['cbsdId'] = device_d['userId']
+    grant_d['operationParam']['maxEirp'] = 20
+    url = GetSasTestHarnessUrl()
+    config = {  'url' : url,
+                'device_a': device_a,
+                'device_b': device_b,
+                'device_c': device_c,
+                'device_d': device_d,
+                'grant_a': grant_a,
+                'grant_b': grant_b,
+                'grant_c': grant_c,
+                'grant_d': grant_d,
+                'ppa_a': ppa_a,
+                'esc_sensor': esc_sensor,
+                'certificateHash': "will provide in the future"
+
+                }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_FAD_2_default_config)
+  def test_WINNF_FT_S_FAD_2(self, config_filename):
+    config = loadConfig(config_filename)
+    sas_th_url = config['url']
+    device_a = config['device_a']
+    device_b = config['device_b']
+    device_c = config['device_c']
+    device_d = config['device_d']
+    grant_a = config['grant_a']
+    grant_b = config['grant_b']
+    grant_c = config['grant_c']
+    grant_d = config['grant_d']
+    ppa_a = config['ppa_a']
+    esc_sensor = config['esc_sensor']
+    # Start the server
+    self._sas_server.startServer()
+
+    # Step 3: Whitelist the FCCID and UserID for the device_b in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_b['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_b['userId']})
+
+    # Step 4: load the device_b with registration to SAS UUT
+    request = {'registrationRequest': [device_b]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    cbsd_id_b = response['cbsdId']
+    # Check registration response of device_b
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+    # Step 5: Load the device_b with grant to SAS UUT
+    request = {'grantRequest': [grant_b]}
+    response = self._sas.Grant(request)['grantResponse'][0]
+    grant_id_b = response['grantId']
+    # Check grant response of device_b
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+    # Whitelist the FCCID and UserID for the device_d in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_d['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_d['userId']})
```

Comment:
but don't you think injecting fccId and user Id are important step even for the CBSDs of other SAS? because this step simulate the synchronization with Fcc DB, so the SAS UUT may refuse the Cbsds in the dump of peer SAS if their fccId and user Id are unknown for the SAS UUT, please let me know your opinion.